src/main/java/com/google/api/codegen/transformer/java/JavaSchemaTypeNameConverter.java

```diff
@@ -0,0 +1,247 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.java;
+
+import com.google.api.codegen.config.FieldConfig;
+import com.google.api.codegen.discovery.Document;
+import com.google.api.codegen.discovery.Node;
+import com.google.api.codegen.discovery.Schema;
+import com.google.api.codegen.discovery.Schema.Type;
+import com.google.api.codegen.transformer.SchemaTypeNameConverter;
+import com.google.api.codegen.util.TypeName;
+import com.google.api.codegen.util.TypeNameConverter;
+import com.google.api.codegen.util.TypedValue;
+import com.google.api.codegen.util.java.JavaTypeTable;
+import com.google.common.base.Strings;
+
+/** The Schema TypeName converter for Java. */
+public class JavaSchemaTypeNameConverter implements SchemaTypeNameConverter {
+
+  /** The package prefix protoc uses if no java package option was provided. */
+  private static final String DEFAULT_JAVA_PACKAGE_PREFIX = "com.google.protos";
```

Comment:
Changing to "com.google.discovery", which is the package for the Google Discovery module. Any other name suggestions welcome.